### DISCUSSION TITLE
Why NSFW is not starting when opening a diff editor from the Git view

### DISCUSSION THREAD

**Discussion by lmcbout** [eclipse-theia/theia#7960] [General]
Why "NSFW" is not starting when I open the diff editor from the Git view ?
I noticed the diff editor from Git view does not trigger a watch file, but from the same Git view, If I open the file instead of opening the diff editor, then the "NSFW" is initiated.
There must me some reasons for it.

**Comment by vince-fugnitto** [comment]
@lmcbout **nsfw** watches for actual files on the filesystem, the diff-editor does not exist on the filesystem (it also does not have a `file` scheme`) so there is no reason for `nsfw` to watch the diff-editor as a whole.

**Reply by akosyakov** [reply]
It depends. Diff editors has 2 models original and modified. File watching is enabled only for `file` protocol. Obviously data read from `git` does not follow `file` protocol and cannot be watched, but in cases for instances when you compare the working copy with the previous revision, file watching will be started for the working copy.
